src/main/java/com/waverleysoftware/listemers/TestListener.java

```diff
@@ -1,31 +1,41 @@
 package com.waverleysoftware.listemers;
 
 import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
 import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestResult;
+import org.testng.internal.TestResult;
 
-@SuppressWarnings("JavadocType")
+@SuppressWarnings({"JavadocType", "unchecked"})
 @Slf4j
 public class TestListener implements IInvokedMethodListener {
+    private static ThreadLocal thread = new ThreadLocal();
 
     @Override
     public void beforeInvocation(final IInvokedMethod method, final ITestResult testResult) {
+        final SoftAssertions softAssert = new SoftAssertions();
+        thread.set(softAssert);
```

Comment:
try to give it more obvious name, as it's not a thread... it's a thread local container for some object you're passing to it...